src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -1098,6 +1103,59 @@ public void closeChannel() {
     }
   }
 
+  /**
+   * Asynchronously closes a channel associated to this partition. Any {@link SFException} occurred
+   * is swallowed and a successful {@link CompletableFuture} is returned instead.
+   */
+  public CompletableFuture<Void> closeChannelAsync() {
+    return closeChannelWrapped()
+        .thenAccept(__ -> onCloseChannelSuccess())
+        .exceptionally(this::tryRecoverFromCloseChannelError);
+  }
+
+  private CompletableFuture<Void> closeChannelWrapped() {
+    try {
+      return this.channel.close();
+    } catch (SFException e) {
+      // Calling channel.close() can throw an SFException instead of putting it into a completable
+      // future.
+      CompletableFuture<Void> future = new CompletableFuture<>();
```

Comment:
this comment is a bit odd - it took me a few reads to understand your point. 
i'd either get rid of it completely - i think it is clear that the sync part of the channel.close can throw or rephrase it to something like "wrap SFException into CompletableFuture to keep consistent method chain"